module_utils/ns1.py

```diff
@@ -63,6 +63,9 @@ def _build_ns1(self):
             self.config["endpoint"] = self.module.params["endpoint"]
         if self.module.params["ignore_ssl"]:
             self.config["ignore-ssl-errors"] = self.module.params["ignore_ssl"]
+        del self.module.params["apiKey"]
+        del self.module.params["endpoint"]
+        del self.module.params["ignore_ssl"]
```

Comment:
Are we cool with adding mutation of module params to a function that is intended to build the SDK config?  Seems like a surprising side effect.

My eventual intent was to pull the sanitization function out of `ns1_zone` and in to this class so all modules could share it.  If we don't think we can generalize sanitization stuff enough to do that, I think it might be more obvious to put these three ops into their own dedicated sanitization function.

Reply:
I'm definitely cool with eating these params so library code doesn't have to worry about it, but I'm not thrilled about my proposal, which is just optimized for minimal changes really. would love to chat it out and get somewhere we're both firmly positive about